src/app/api/agencies/[id]/route.ts

```diff
@@ -0,0 +1,34 @@
+import { getAgencyById } from '@/server/actions/Agencies';
+import { JSendResponse } from '@/utils/types';
+
+export async function GET(
+  request: Request,
+  { params }: { params: { id: string } }
+) {
+  const id = params.id;
+
+  try {
+    const agency = await getAgencyById(id);
+
+    return Response.json(
+      new JSendResponse({
+        status: 'success',
+        data: { agency: agency },
+      }),
+      { status: 200 }
+    );
+  } catch (error) {
+    if (error instanceof JSendResponse) {
```

Comment:
duplicate condition

you should only need the one below so that non-JSend errors can indicate a server error

Reply:
Good catch! Fixed in 9cb9884